### ISSUE TITLE
Config is not a string but a hash

### ISSUE THREAD

**Issue by MCyprien** [voxpupuli/puppet-logrotate#77]
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: 4.10.7
- Ruby: 2.3.0
- Distribution: Rhel6
- Module version: 3.0.0

## How to reproduce (e.g Puppet code you use)

```
class { '::logrotate':
  ensure => 'latest',
  config => {
    dateext  => true,
    compress => true,
  }
}
```

## What are you seeing

` Error: Evaluation Error: Error while evaluating a Resource Statement, Class[Logrotate]: parameter 'config' expects a value of type Undef or String, got Struct at ...`

## What behaviour did you expect instead

Compilation of catalog

## Any additional information you'd like to impart

in your init.pp :
```
class logrotate (
...
  Optional[String] $config           = undef,
...
)

```
Hope I could help you :)

**Comment by bastelfreak** [comment]
Hi @MCyprien, thanks for rising this issue. Can you try this:

```puppet
class { '::logrotate':
  ensure => 'latest',
  config => {
    'dateext'  => true,
    'compress' => true,
  }
}
```

**Comment by MCyprien** [comment]
Hi @bastelfreak,

I tried with with quotes but no change on the issue. It still doesn't compile. :(

**Comment by bastelfreak** [comment]
Ah stupid me. There is an error in our init.pp, the datatype should be Hash and not String. I will fix that later.

**Comment by bastelfreak** [comment]
This will be fixed in https://github.com/voxpupuli/puppet-logrotate/pull/78

**Comment by MCyprien** [comment]
Nice it was resolved with 2.3.1 Thanks
